### ISSUE TITLE
Switch to GitHub GraphQL API v4

### ISSUE THREAD

**Issue by erezrokah** [netlify/netlify-cms#2431]
Netlify cms uses [GitHub v3 REST API](https://developer.github.com/v3/).

Using the REST API generate numerous requests even for simple operations.

e.g. saving a post in editorial workflow:
![image](https://user-images.githubusercontent.com/26760571/60881612-bb000400-a24e-11e9-8278-9874e98db9a7.png)

(there are more requests not present in the screenshot).

Also, the GitHub api returns a lot of data that is not necessarily required by the CMS.

The only reference for switching to [GraphQL](https://developer.github.com/v4/guides/forming-calls/) I could find is [here](https://github.com/netlify/netlify-cms/issues/320)

I don't mind to start working on a PR (I think we can have it under a flag and gradually replace API calls).

**Comment by Benaiah** [comment]
@erezrokah while we can't move to the GraphQL completely, as it doesn't support all the mutations we need (at least when I last checked), it's still definitely a major opportunity to cut down on numbers of requests and request sizes when _getting_ data. I'd definitely be in favor of gradually replacing REST calls with GraphQL when appropriate, but I suspect we'll need some REST calls for certain mutation operations even after we do so.

**Comment by erezrokah** [comment]
@Benaiah Thank you for the quick reply.

I've already started playing with the GraphQL API.
https://github.com/erezrokah/netlify-cms/tree/github_graphql_api

I made it in a way both APIs co-exist and the GraphQL API falls back to the REST API for unsupported operations.

If you want to test it you'll need to set `use_graph_ql: true` under `backend` in `config.yml`

It is still work in progress, and will need some tests/code clean.

It doesn't leverage all the querying capabilities to minimize requests completely though (I think it will require a more significant change)
